src/framework/auth/providers/firebase-auth.provider.ts

```diff
@@ -0,0 +1,220 @@
+          this.processFailure(res, this.getConfigValue('login.redirect.failure'), [res.message]),
+        );
+      });
+  }
+
+  /**
+   * Firebase registration.
+   *
+   * @param data any
+   * @returns Observable<NbAuthResult>
+   */
+  register(data?: any): Observable<NbAuthResult> {
+    return Observable.fromPromise(firebase.auth().createUserWithEmailAndPassword(data.email, data.password))
+      .map((res) => {
+        return Observable.fromPromise(firebase.auth().currentUser.updateProfile({
+          displayName: data.fullName,
+          photoURL: '',
+        })).map((update) => {
+          return this.processSuccess(res, this.getConfigValue('register.redirect.success'), [res.message]);
+        });
+      })
+      .catch((res) => {
+        return Observable.of(
+          this.processFailure(res, this.getConfigValue('register.redirect.failure'), [res.message]),
+        );
+      });
+  }
+
+  /**
+   * Firebase restore password.
+   *
+   * @param data any
+   * @returns Observable<NbAuthResult>
+   */
+  requestPassword(data?: any): Observable<NbAuthResult> {
+    return Observable.fromPromise(firebase.auth().sendPasswordResetEmail(data.email))
+      .map((res) => {
+        return this.processSuccess(res, this.getConfigValue('requestPass.redirect.success'), []);
+      })
+      .catch((res) => {
+        return Observable.of(this.processFailure(res,  this.getConfigValue('requestPass.redirect.failure'),
+          [res.message]));
+      });
+  }
+
+  /**
+   * Firebase reset password.
+   *
+   * @param data any
+   * @returns Observable<NbAuthResult>
+   */
+  resetPassword(data?: any): Observable<NbAuthResult> {
+    if (firebase.auth().currentUser) {
+      return Observable.fromPromise(firebase.auth().currentUser.updatePassword(data.password))
+        .map((res) => {
+          return this.processSuccess(res, this.getConfigValue('resetPass.redirect.success'), []);
+        })
+        .catch((res) => {
+          return Observable.of(this.processFailure(res,  this.getConfigValue('resetPass.redirect.failure'),
+            [res.message]));
+        });
+    }
+
+    return Observable.of(this.processFailure([],  this.getConfigValue('resetPass.redirect.failure'),
+      ['Please, sign in to be able to reset your password']));
+  }
+
+  /**
+   * Firebase logout.
+   *
+   * @param data any
+   * @returns Observable<NbAuthResult>
+   */
+  logout(data?: any): Observable<NbAuthResult> {
+    return Observable.fromPromise(firebase.auth().signOut())
+      .map((res) => {
+        return this.processSuccess(res, this.getConfigValue('logout.redirect.success'),  []);
+      })
+      .catch((res) => {
+        return Observable.of(this.processFailure(res, this.getConfigValue('logout.redirect.failure'),
+          [res.message]));
+      });
+  }
+
+  private processSuccess(response?: any, redirect?: any, messages?: any): NbAuthResult {
+    return new NbAuthResult(true, response, redirect, [], messages);
```

Comment:
should add the last parameter rawToken, otherwise tokenService ignores this payload and does not store it.

Reply:
infact, since this doesn't return a JWT. we need to store `JSON.stringify(response)` as the rawToken